docs/developer-guides/versioned-plutus-data-migration-guide.md

```diff
@@ -0,0 +1,69 @@
+# Versioned Plutus Data - Migration Guide
+
+## About
+
+This document describes steps needed to migrate a running chain to use the new smart contracts
+that introduce versioned Plutus data.
+
+## Context
+
+To make it easier for Partner Chains to keep up with the evolution of the Partner Chains SDK
+while avoiding chain resets and other disruptions, the new Partner Chains Smart Contracts version
+introduces a new layout of on-chain data in the form of *generic versioned Plutus data*. This
+means that every datum attached to UTxO-s produced by the Partner Chains tooling and observed by
+the node, now follows a common structure with:
+- **smart contract-specific data section** with a schema fixed by the smart contract version
+- **generic data section**, whose schema can freely evolve without smart contract changes
+- **version number**
+
+## Migration steps - Ariadne
+
+At the moment, the only feature actively supported by the Partner Chains SDK which uses Plutus
+data observability is the Ariadne committee selection mechanism, which observes the following
+data:
+- D-Parameter
+- Permissioned candidates list
+- Cardano SPO registrations
```

Comment:
This guide will be for v1.4, when in v1.3 we introduce observability of token release from reserve to illiquid supply. Will this script be changed as well?

Reply:
No, native tokens is a completely new mechanism so no migration needed. There will be no legacy version, but the initial implementation can support versioning from the start.